### ISSUE TITLE
"Didn't find any devices" with VPN enabled

### ISSUE THREAD

**Issue by gabemunro** [delfick/photons-interactor#14]
Thank you so very much for this library first off. You're a hero for sharing this with us.

I have been running photons-interactor without issue on my macOS system at home, and just yesterday added a layer of complication to things by adding ExpressVPN to the mix. 

Of course, before connecting to the VPN things work fine. As soon as I connect, the log starts showing the string of "Didn't find any devices" errors. Once I disconnect, within seconds photons-interactor detects my Beam. Log is below. Between the error and where it finds the beam, I had disabled ExpressVPN.

My first idea was to add the version of python used by photons-interactor to the list of apps to not be routed through ExpressVPN, but for some reason the client app won't let me. I'm just wondering if you have any ideas off the top of your head on what specifically I should be trying to whitelist/create an exception for, or if there's some other obvious solution that I'm missing. 

Thank you in advance. I genuinely appreciate any guidance/suggestions you could provide.

`2020-06-17 11:03:50,667 ERROR   photons_app.helpers "Didn't find any devices"
Traceback (most recent call last):
  File "/Users/pathtophotons/.interactor/lib/python3.7/site-packages/photons_control/script.py", line 410, in getter
    msg = await gen.asend(complete)
  File "/Users/pathtophotons/.interactor/lib/python3.7/site-packages/photons_control/script.py", line 219, in gen
    reference, args_for_run, timeout=kwargs.get("find_timeout", 20)
  File "/Users/pathtophotons/.interactor/lib/python3.7/site-packages/photons_control/script.py", line 36, in find_serials
    found, serials = await reference.find(args_for_run, timeout=timeout)
  File "/Users/pathtophotons/.interactor/lib/python3.7/site-packages/photons_app/special.py", line 80, in find
    return await self.found
  File "/Users/pathtophotons/.interactor/lib/python3.7/site-packages/photons_app/helpers.py", line 540, in __await__
    return (yield from self.info["fut"])
  File "/Users/pathtophotons/.interactor/lib/python3.7/site-packages/photons_app/special.py", line 95, in find_serials
    return await afr.find_devices(timeout=timeout, broadcast=broadcast, raise_on_none=True)
  File "/Users/pathtophotons/.interactor/lib/python3.7/site-packages/photons_transport/comms/base.py", line 233, in find_devices
    found, _ = await self.find_specific_serials(None, **kwargs)
  File "/Users/pathtophotons/.interactor/lib/python3.7/site-packages/photons_transport/comms/base.py", line 241, in find_specific_serials
    found = await self._find_specific_serials(serials, **kwargs)
  File "/Users/pathtophotons/.interactor/lib/python3.7/site-packages/photons_transport/comms/base.py", line 259, in _find_specific_serials
    raise FoundNoDevices()
photons_app.errors.FoundNoDevices: "Didn't find any devices"
2020-06-17 11:04:08,560 INFO    photons_transport.transports.udp Creating datagram endpoint	address=('xxx.xxx.xxx.xxx', 56700)	serial=xxxxxxxxx`

**Comment by gabemunro** [comment]
One other curious addition - I also tried restricting the VPN to only specific apps (essentially browsers only), and as soon as I enabled ExpressVPN photons-interactor started throwing the "Didn't find any devices" error, despite the new restriction.

**Comment by Djelibeybi** [comment]
VPNs tend to have really bad effects on local broadcasts which is how Photons discovers lights. You _may_ be able to bypass this by hardcoding discovery (which requires fixed IP addresses for each bulb) but you'd still need to work out how to properly exclude Photons from the VPN itself.

**Comment by delfick** [comment]
Hello,

This annoys me too when I turn on my VPN at work!

The solution is to tell Photons to use a more specific broadcast address, because the default `255.255.255.255` will end up going to your VPN.

The current documentation is a complete mess and it isn't very obvious how to do this. The good news is the new documentation is a lot better and (imho) is a bit easier to follow. I am another month or so away from releasing it (and a bunch of [bunch](https://github.com/delfick/photons-core/pull/11) of [ridiculous](https://github.com/delfick/photons-core/commit/f97007713a2c1ba4e441476ec26d7c388f748288) [changes](https://github.com/delfick/photons-core/commit/3d83afb79b57ca2f932c17d153a81a7ba352cdd1)) but for now you can view it on github which is close enough (the rendered docs will have better links and looks and such)

[This page](https://github.com/delfick/photons-core/blob/master/docs/start/configuration/index.rst ) explains where the configuration is and the Targets section explains how to change the default broadcast address.

The [hard coded discovery](https://github.com/delfick/photons-core/blob/master/docs/start/configuration/discovery_options.rst ) Avi mentions is also a solution but the easiest way to get around the VPN is to just change the broadcast address (hard coded discovery doesn't work so well if it references a device that is turned off for example, which I'll fix in the future sometime)

**Comment by gabemunro** [comment]
THANK YOU! So happy to have such quick and helpful responses from you both. I got it working by swapping the 255.255.255.255 address for one on my network and it's back to working.

The only Lifx product I have is this Beam I just received, along with about 50 other HomeKit devices (I'm deeeeep in the ecosystem), and was beyond disappointed that I couldn't trigger any effects with HomeKit like I was able to with my Nanoleaf Light Panels. Between Photons-Interactor, Homebridge, and Homebridge-HTTP-Switch, I'm finally able to trigger things using Siri on my HomePods, and include some basic animation in scenes.

Avi - I'm not at all well versed in any languages outside of basic HTML, but through your blog posts about this and how you were configuring your Flic buttons, I was able to cobble together a solution.

Again, super thankful, and super excited to see what comes next for this.
